Source/WTF/wtf/threads/Signals.cpp

```diff
@@ -203,6 +206,157 @@ kern_return_t catch_mach_exception_raise_state_identity(mach_port_t, mach_port_t
     return KERN_FAILURE;
 }
 
+#ifdef EXCEPTION_IDENTITY_PROTECTED
+
+static thread_act_t threadIDGetThreadPort(uint64_t threadID)
+{
+    kern_return_t kr;
+    mach_msg_type_number_t i, thcnt;
+    thread_act_array_t threads;
+    struct thread_identifier_info tiinfo;
+    unsigned infoCount = THREAD_IDENTIFIER_INFO_COUNT;
+    thread_act_t thread = THREAD_NULL;
+
+    task_threads(mach_task_self(), &threads, &thcnt);
+    for (i = 0; i < thcnt; i++) {
+        if (thread != THREAD_NULL) {
+            // clean up all the unnecessary thread ports like a good citizen
+            ASSERT(thread != threads[i]);
+            (void)mach_port_deallocate(mach_task_self(), threads[i]);
+            continue;
+        }
+
+        kr = thread_info(threads[i], THREAD_IDENTIFIER_INFO,
+        (thread_info_t) &tiinfo, &infoCount);
+
+        if (kr == MACH_SEND_INVALID_DEST) {
+            // ignore threads that have been destroyed
+            continue;
+        }
+
+        if (tiinfo.thread_id == threadID) {
+            // we found the thread we are looking for
+            thread = threads[i];
+            continue;
+        }
+        (void)mach_port_deallocate(mach_task_self(), threads[i]);
+    }
+    return thread;
+}
+
+kern_return_t catch_mach_exception_raise_state(
+    mach_port_t,
+    exception_type_t,
+    const mach_exception_data_t,
+    mach_msg_type_number_t,
+    int*,
+    const thread_state_t,
+    mach_msg_type_number_t,
+    thread_state_t,
+    mach_msg_type_number_t*)
+{
+    dataLogLn("We should not have called catch_mach_exception_raise_state, please file a bug at bugs.webkit.org");
+    return KERN_FAILURE;
+}
+
+kern_return_t catch_mach_exception_raise_identity_protected(
+    mach_port_t exceptionPort,
+    uint64_t threadID,
+    mach_port_t taskIDToken,
+    exception_type_t exceptionType,
+    mach_exception_data_t exceptionData,
+    mach_msg_type_number_t dataCount)
+{
+    UNUSED_PARAM(threadID);
+    UNUSED_PARAM(taskIDToken);
+
+    kern_return_t kr;
```

Comment:
WebKit style doesn't forward declare variables like this. We'd normally declare it at the first assignment unless there's some other compelling reason to have it up front. Also, ditto on naming.

Reply:
On the naming bit, `kr` is the standard convention for this variable, and is already named as such extensively throughout this file. I think it'd be more confusing to use something else in this case.